File: tests/test_compat.py
"""Conformance run of dev_server.py:_compat_satisfies against the shared vector.
Its twin (tests/test_compat.js) runs the SAME vector against js/core/compat.js —
both must pass for any change to the compat semantics."""
import json
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parents[1]))
import dev_server as ds

vector = json.loads((Path(__file__).parent / "compat-vector.json").read_text(encoding="utf-8"))
failed = 0
for case in vector["cases"]:
    ok, reason = ds._compat_satisfies(case["platform"], case["decl"])
    if ok != case["expect"]:
        failed += 1
        print(f"  FAIL platform={case['platform']!r} decl={case['decl']!r} "
              f"→ {ok} (expected {case['expect']}) [{case['why']}] reason={reason}")
if failed:
    print(f"{failed}/{len(vector['cases'])} CASES FAILED")
    sys.exit(1)
print(f"ALL {len(vector['cases'])} COMPAT CASES PASSED (python)")

File: tests/test_compat.js
/* Conformance run of js/core/compat.js against the shared vector (node).
   Twin of tests/test_compat.py — both must pass identically. */
'use strict';
const fs = require('fs');
const path = require('path');

const Compat = require(path.join(__dirname, '..', 'js', 'core', 'compat.js'));
const vector = JSON.parse(fs.readFileSync(path.join(__dirname, 'compat-vector.json'), 'utf-8'));

let failed = 0;
for (const c of vector.cases) {
  const { ok, reason } = Compat.satisfies(c.platform, c.decl);
  if (ok !== c.expect) {
    failed++;
    console.log(`  FAIL platform=${JSON.stringify(c.platform)} decl=${JSON.stringify(c.decl)}` +
      ` → ${ok} (expected ${c.expect}) [${c.why}] reason=${reason}`);
  }
}
if (failed) {
  console.log(`${failed}/${vector.cases.length} CASES FAILED`);
  process.exit(1);
}
console.log(`ALL ${vector.cases.length} COMPAT CASES PASSED (js)`);
